### PR TITLE
REST API v2.1.0 documentation

### PR DESCRIPTION
**Pull request by abellotti**
REST API v2.1.0 documentation.

@mfeifer please review.

Marked as WIP as I want to squash commits before merging.  Just thought it may be easier to review.

### CODE REVIEW

File: api/appendices/provider_types.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
```
Comment (Fryguy) [inline api/appendices/provider_types.adoc]: I thought leading blanks in the files caused problems

Comment (abellotti) [inline api/appendices/provider_types.adoc]: actually, the leading blanks in the files fix the problem asciibinder had with the [[ ]] tagged entries where they were shown in the rendered html.

File: api/appendices/resource_attributes.adoc [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+[cols="1<,1<,3<",options="header",width="100%"]
+|=====================
+| Attribute | Type | Description
+| name | string | Name of tag
+| description | string | Description of tag
+| category | hash | Category identifier for which the tag is getting added to, requires an *href*, *id* or *name* attribute in the hash
+|=====================
+
+When adding via the *assign* action to /api/categories/:id/tags, the category should not be specified
```
Comment (imtayadeway) [inline api/appendices/resource_attributes.adoc]: Not sure I understand this. Is it not the `create` action?

File: api/overview/auth.adoc [hunk 3]
```diff
@@ -67,8 +66,8 @@ Content-Type: application/json
 [source,sh]
 ----
 $ curl -i -X GET -H "Accept: application/json"
-       -H "X-Auth-Token: af0245-238722-4d23db"
-       http://localhost:3000/api/services/1013
+       -H "X-Auth-Token: 93d98fd44cc5a401ff385ba87b21fd13"
+       http://localhost:300/api/services/1013
```
Comment (imtayadeway) [inline api/overview/auth.adoc]: Is this really port 300?

File: api/overview/auth.adoc [hunk 4]
```diff
@@ -85,3 +84,50 @@ When a request fails due to an invalid token, the client would then need
...
+
+{
+  "auth_token": "8789bcd6562beacbaf9f77927c2825e2",
+  "token_ttl": 3600,
+  "expires_on": "2015-11-02T23:24:03Z"
+}
+----
+
+[[api-token-ttl-configuration]]
+==== Configuing the REST API Token Duration
```
Comment (imtayadeway) [inline api/overview/auth.adoc]: s/Configuing/Configuring/

File: api/overview/authorization.adoc [hunk 5]
```diff
@@ -0,0 +1,72 @@
+
+[[rest-api-authorization]]
+=== REST API Authorization
+
+When authenticating to the appliance, the current group of the authenticated user will be used for
+determining the authorization of what the user can see and do using the REST API. While this
+is acceptable when users only belong to a single group, this falls short when users are authenticating
+using their External Authentication credentials (IPA, Ldap or AD) where they may belong to
```
Comment (imtayadeway) [inline api/overview/authorization.adoc]: Not sure, but should Ldap be uppercase?

File: api/reference/categories_tags.adoc [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+[NOTE]
+====
+Please refer to the link:../appendices/resource_attributes.html#categories[Resource Attributes]
+page for a list of available attributes when creating Categories.
+====
+
+[[creating-tags]]
+=== Creating Tags
+
+New tags for a category can be created as either as a tag collection post or via a tag
```
Comment (imtayadeway) [inline api/reference/categories_tags.adoc]: maybe simply "[...] created either as a tag [...]"

File: api/reference/categories_tags.adoc [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+[NOTE]
+====
+Please refer to the link:../appendices/resource_attributes.html#tags[Resource Attributes]
+page for a list of available attributes when creating Tags.
+====
+
+[[editing-categories-tags]]
+=== Editing Categories & Tags
+
+Deleting Categories and Tags can be done via the *edit* action:
```
Comment (imtayadeway) [inline api/reference/categories_tags.adoc]: s/Deleting/Editing/

File: api/reference/service_custom_actions.adoc [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+[[service-custom-actions]]
+== Service Custom Actions
+
+With button groups, custom buttons and related dialogs defined on Service Templates, the REST API provides visibility
+to custom buttons on service templates and related ordered services. 
+
+This is made available with the _custom_actions_ and _custom_action_buttons_ virtual attributes on service_templates
+and services.
+
+* _custom_actions_ return the custom buttons as layed out in the UI on the service. The information returned includes
```
Comment (imtayadeway) [inline api/reference/service_custom_actions.adoc]: s/return/returns/

**Comment by imtayadeway** [comment]
@abellotti this is awesome :fireworks: . Minor comments above

**Comment by abellotti** [comment]
Thanks Tim for the review, I'll update and repush.

**Comment by abellotti** [comment]
Updated and moved examples to separate topic on left hand side for easier browsing of examples by category.

@imtayadeway @mfeifer Let me know if there's anything else to update, if not, I'll squash/repush/unwip.  Thanks.

**Comment by imtayadeway** [comment]
@abellotti :+1: :cake: :musical_keyboard:

**Comment by abellotti** [comment]
@mfeifer anything to change ? can this be merged ?  Thanks.

**Pull request merged**